src/coresyftools/coresyftools/tests/test_tool.py

```diff
@@ -241,3 +241,49 @@ def run(self, bindings):
         cmd = '--input f1 --output f2 --param astr'.split()
         self.assertRaises(EmptyOutputFile, lambda: tool.execute(cmd))
         os.remove('f1')
+
+
+    def test_invoke_shell_command(self):
+        coresyf_tool = CoReSyFTool(self.runfile)
+        (pipeline, stdout, stderr) = coresyf_tool.invoke_shell_command("test {op}", op="1")
+        self.assertEqual(pipeline.returncode, 0)
+        (pipeline, stdout, stderr) = coresyf_tool.invoke_shell_command("test {op1} = {op2}", op1="1", op2="2")
+        self.assertEqual(pipeline.returncode, 1)
+        (pipeline, stdout, stderr) = coresyf_tool.invoke_shell_command("echo -n out")
+        self.assertEqual(stdout.read(), 'out')
+        (pipeline, stdout, stderr) = coresyf_tool.invoke_shell_command("rm nofile")
+        self.assertIn('rm: cannot remove', stderr.read())
+
+    
+    def _extend_manifest(self, extra_fields):
+        manifest = self.manifest.copy()
+        manifest.update(extra_fields)
+        with open('manifest.json', 'w') as manifest_file:
+            json.dump(manifest, manifest_file)
+        self.runfile = os.path.join(os.getcwd(), 'run')
+
+    def test_can_run_shell_command_of_manifest(self):
+        self._extend_manifest({
+            'command': 'cp {input} {output}; echo {param}'
+        })
+        with open('f1', 'w') as f1:
```

Comment:
I know this is a test file, but variables named like this is a knot in the brain when reading the code. `file1`, `file2`.

Reply:
Yes, you are right. But, come on, here you only have to look to the line above to discover what is the variable, this is a single line `with` block scope :stuck_out_tongue:

Reply:
It takes as long to name `f1` as it does to name `file1`. And in the long run it improves code readibility. But it's ok if you leave it like this for now.